src/controllers/trainee/Controller.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response } from 'express';
+import UserRepository from '../../repositories/user/UserRepository';
+import SystemResponse from '../../libs/routes/SystemResponse';
+import { IRequest } from '../../libs/interface';
+import { initSearch } from '../../libs/utils/helper';
+
+class TraineeController {
+  static instance: TraineeController;
+  public userRepo = new UserRepository();
+  private systemResponse: SystemResponse = new SystemResponse();
+  static getInstance(): TraineeController {
+    if (TraineeController.instance instanceof TraineeController) {
+      return TraineeController.instance;
+    }
+    return (TraineeController.instance = new TraineeController());
+  }
+  private constructor() { }
+
+  create = async (req: IRequest, res: Response): Promise<Response> => {
+    try {
+      const userId = req.user.userId;
+      const data = req.body;
+      const record = { ...data, userId };
+      const result = await this.userRepo.create(record);
+      if (result) {
+        delete result._id;
+        return this.systemResponse.success(
+          res,
+          `Trainee added Successfully`,
+          200,
+          result
+        );
+      }
+      return this.systemResponse.success(res, `Unauthorized User`, 403, result);
+    } catch (err) {
+      return this.systemResponse.failure(res, err.message, 500, err);
+    }
+  };
+
+  list = async (req: IRequest, res: Response): Promise<Response> => {
+    try {
+      const { skip, limit, sort, search, ...query } = req.query;
+      const options: object = { skip, limit, sort };
+      const filter: object = await initSearch(search);
+      let result;
+      const querySchema = Object.keys(filter).length ? filter : query;
+      result = await this.userRepo.getAllRecord(querySchema, options);
+      if (result.length !== 0) {
```

Comment:
result.length !== 0 can be written as result.length

Reply:
Done